### PR TITLE
feat: add support for customizing telegraf image

### LINKED ISSUE #11: Make telegraf sidecar image configurable either by flag and annotation with sane default
_Describe the issue here._

Right now the telegraf image to use is hardcoded in the source code.

I'd like this image to be change by a deployment flag, like `--telegraf-image=docker.io/library/telelgraf:1.13` and also have an annotation `telegraf.influxdata.com/image` to be used to change the image to be used on a pod basis

### PR DESCRIPTION
**Pull request by wojciechka**
Closes #11

Allows passing telegraf image as command line option as well as using an annotation.

- [x] Signed the [InfluxData CLA](https://www.influxdata.com/legal/cla/)
  ([if necessary](https://github.com/influxdata/telegraf-operator/blob/master/CONTRIBUTING.md#sign-the-influxdata-cla))
- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -37,6 +37,10 @@ var (
 	setupLog = ctrl.Log.WithName("setup")
 )
 
+const (
+	defaultTelegrafImage = "docker.io/library/telegraf:1.12"
```
Comment (gitirabassi) [inline main.go]:
```suggestion
	defaultTelegrafImage = "docker.io/library/telegraf:1.13"
```

File: sidecar_test.go [hunk 2]
```diff
@@ -193,10 +196,105 @@ spec:
...
+
+			if tt.wantPod != "" {
+				if want, got := strings.TrimSpace(tt.wantPod), strings.TrimSpace(toYAML(t, tt.pod)); got != want {
+					fmt.Printf("WTF\n%s\n", got)
+					t.Errorf("unexpected pod got:\n%v\nwant:\n%v", got, want)
+				}
+			}
+		})
 	}
+
```
Comment (gitirabassi) [inline sidecar_test.go]: Maybe remove the commented code if not used

Comment (wojciechka) [reply sidecar_test.go]: OMG, I completely forgot about that. Removed.

**Comment by wojciechka** [comment]
> The only strictly needed change would be the additional annatation in the README.md in the root of the repository

Good catch. I forgot the README includes list of options. Done. Also removed unused code.

**Pull request merged**